### PR TITLE
add the convention for Splunk 'source' field

### PR DESCRIPTION
**Pull request by rockb1017**
**Description:** 
`source` field in Splunk typically is the file name. https://docs.splunk.com/Documentation/Splunk/latest/Data/Aboutdefaultfields
Right now, it is using `service.name`, field as source. 
O11y platform has a different use case for `service.name` field. So it is creating a conflict in building a unified helm chart. 
So instead, I am suggesting to use `com.splunk.source`.

@dmitryax Please provide any more information if i have missed anything or suggest better approach.

### CODE REVIEW

**Comment by dmitryax** [comment]
Thanks! I support this change.

Please also make according change to [the logs specification](https://github.com/open-telemetry/opentelemetry-specification/blob/938c30078d51560f04b7125d43056052a3a9742d/specification/logs/data-model.md#splunk-hec) first/

Then, once the specification change is merged, please do the following updates to this PR:
- the same mapping should be changed in [Splunk HEC receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/splunkhecreceiver)
- add a line to the CHANGELOG marked as a BREAKING CHANGE

@tigrannajaryan @atoulme please let us know if you support this change?

**Comment by atoulme** [comment]
Right, the spec needs to change first. I just followed the spec for the mapping. I am ok with this change, however, how would service.name be determined from a HEC event now?

**Comment by rockb1017** [comment]
PR created for log specification. https://github.com/open-telemetry/opentelemetry-specification/pull/1861

**Comment by tigrannajaryan** [comment]
> So instead, I am suggesting to use `com.splunk.source`.

Sounds reasonable and the name is good, it matches the requirements for vendor-specific attribute names.

Question: who do we expect do set this attribute? Will it be for example a user-defined attributes processor that copies the `file.name` attribute into `com.splunk.source` or for example `filelog` receiver will be configured to put the file name directly into  `com.splunk.source`?

**Comment by tigrannajaryan** [comment]
One more: we also need a corresponding change in HEC receiver: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1c2f2668f308ad87330fd1a97de9e6383ecab744/receiver/splunkhecreceiver/splunk_to_logdata.go#L58

This will ensure passthrough when receiving and exporting HEC protocol works correctly and keeps the source unchanged.

**Comment by rockb1017** [comment]
So before sending data to `splunkhecexporter`, user would need to add a processor like this 
```
  resource/splunk:
    attributes:
    - key: host.name
      from_attribute: k8s.node.name
      action: upsert
    - key: com.splunk.source
      from_attribute: filename
      action: upsert
    - key: com.splunk.sourcetype
      from_attribute: k8s.pod.annotations.splunk.com/sourcetype
      action: upsert
    - key: com.splunk.index
      from_attribute: k8s.namespace.annotations.splunk.com/index
      action: upsert
    - key: com.splunk.index
      from_attribute: k8s.pod.annotations.splunk.com/index
      action: upsert
```

**Comment by rockb1017** [comment]
it is hard coded as `service.name` and this PR is to change it to `com.splunk.source`. 
for users to have correct data mapping for Splunk default fields, it means they all must have that specific processor. 

I think a better approach would be supporting key field parameters like `source_key` in `splunkhecexporter`. it would be by default `com.splunk.source`. 

if a record doesn't have a attribute of `com.splunk.source` (source_key value), then default source value is used (`source` parameter) 

I have created issue a while ago here. 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2653

**Comment by atoulme** [comment]
I think your issue might need to be implemented first then. It seems less impacting. I asked earlier `how would service.name be determined from a HEC event now?`

**Comment by rockb1017** [comment]
>  `how would service.name be determined from a HEC event now?`

Hey, I didn't understand your question. Why would `service.name` need to be determined from a HEC event ?

**Comment by atoulme** [comment]
It is because a HEC event representing a log would eventually be translated to a log record in the o11y model. The log record would then be consumed by a variety of exporters. It can be consumed by a HEC exporter. It could also be consumed by an Otel gRPC exporter to go straight to a service. In that case, the service would rightfully expect the metadata of the log record to contain a value for service.name, and would use that to associate the record with other elements from the same service (notably, traces).

**Comment by rockb1017** [comment]
I see, thanks for explanation. 
It is hard to answer. `source` field has not been used to make association with other data, but there is nothing better fitting than source when you think about it. 

by the definition, https://docs.splunk.com/Splexicon:Source
"A default field that identifies the source of an event, that is, where the event originated. In the case of data monitored from files and directories, the source consists of the full pathname of the file or directory. In the case of a network-based source, the source field consists of the protocol and port, such as UDP:514."
I think it can be left as `service.name`. So for k8s container logs, it is also true to have "pod_name/namespace_name" instead of the actual log path because that is where event originated. Let me confirm with our PM. Thanks!

**Comment by atoulme** [comment]
To reiterate, I'm ok with the change. I think it might be great to implement it along the lines of your idea of making source mapping configurable (#2653).

File: Makefile [hunk 1]
```diff
@@ -82,7 +82,8 @@ gofmt:
 
 .PHONY: golint
 golint:
-	$(MAKE) for-all-target TARGET="lint"
+	$(MAKE)
+ TARGET="lint"
```
Comment (tigrannajaryan) [inline Makefile:86]: Why this change?

**Review by tigrannajaryan — COMMENTED** [review]
LGTM, except one comment on what appears to be unintended change.

**Comment by rockb1017** [comment]
right. removed unintended change from my commit

**Review by dmitryax — APPROVED** [review]

**Review by tigrannajaryan — APPROVED** [review]
LGTM.
